arch/Xtensa/XtensaDisassembler.c

```diff
@@ -0,0 +1,298 @@
+/* Capstone Disassembly Engine, http://www.capstone-engine.org */
+/* By Nguyen Anh Quynh <aquynh@gmail.com>, 2013-2022, */
+/*    Rot127 <unisono@quyllur.org> 2022-2023 */
+/* Automatically translated source file from LLVM. */
+
+/* LLVM-commit: <commit> */
+/* LLVM-tag: <tag> */
+
+/* Only small edits allowed. */
+/* For multiple similar edits, please create a Patch for the translator. */
+
+/* Capstone's C++ file translator: */
+/* https://github.com/capstone-engine/capstone/tree/next/suite/auto-sync */
+
+//===-- XtensaDisassembler.cpp - Disassembler for Xtensa ------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements the XtensaDisassembler class.
+//
+//===----------------------------------------------------------------------===//
+
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <capstone/platform.h>
+
+#include "../../MathExtras.h"
+#include "../../MCDisassembler.h"
+#include "../../MCFixedLenDisassembler.h"
+#include "../../SStream.h"
+#include "../../cs_priv.h"
+#include "../../utils.h"
+#include "priv.h"
+
+#define GET_INSTRINFO_MC_DESC
+#include "XtensaGenInstrInfo.inc"
+
+#define CONCAT(a, b) CONCAT_(a, b)
+#define CONCAT_(a, b) a##_##b
+
+#define DEBUG_TYPE "Xtensa-disassembler"
+
+static const unsigned ARDecoderTable[] = {
+	Xtensa_A0,  Xtensa_SP,	Xtensa_A2,  Xtensa_A3, Xtensa_A4,  Xtensa_A5,
+	Xtensa_A6,  Xtensa_A7,	Xtensa_A8,  Xtensa_A9, Xtensa_A10, Xtensa_A11,
+	Xtensa_A12, Xtensa_A13, Xtensa_A14, Xtensa_A15
+};
+
+static DecodeStatus DecodeARRegisterClass(MCInst *Inst, uint64_t RegNo,
+					  uint64_t Address, const void *Decoder)
+{
+	if (RegNo >= sizeof(ARDecoderTable))
+		return MCDisassembler_Fail;
+
+	unsigned Reg = ARDecoderTable[RegNo];
+	MCOperand_CreateReg0(Inst, (Reg));
+	return MCDisassembler_Success;
+}
+
+static const unsigned SRDecoderTable[] = { Xtensa_SAR, 3 };
+
+static DecodeStatus DecodeSRRegisterClass(MCInst *Inst, uint64_t RegNo,
+					  uint64_t Address, const void *Decoder)
+{
+	if (RegNo > 255)
+		return MCDisassembler_Fail;
+
+	for (unsigned i = 0; i + 1 < sizeof(SRDecoderTable); i += 2) {
```

Comment:
> The clang-tidy warnings are a false positive, because SRDecoderTable[i + 1] does not cause the array to go out of bounds if i + 1 < sizeof(SRDecoderTable) is satisfied.

We are not in C++ realm here unfortunately. `sizeof` returns the number of bytes. Not the number of elements.
Please use the macro `ARR_SIZE()` for this.